llvm/lib/Target/ARM/Thumb1FrameLowering.cpp

```diff
@@ -222,7 +222,8 @@ void Thumb1FrameLowering::emitPrologue(MachineFunction &MF,
     case ARM::R8:
     case ARM::R9:
     case ARM::R10:
-      if (STI.splitFramePushPop(MF)) {
+      if (STI.getPushPopSplitVariation(MF) ==
+          ARMSubtarget::PushPopSplitVariation::R7Split) {
```

Comment:
Given this is Thumb1FrameLowering... isn't this always true?

Reply:
It looks like Thumb1FrameLowering is initialized only when isThumb1Only() is true, so the variation will always be R7Split. Therefore the if statement is unnecessary.